helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -112,11 +112,17 @@ spec:
         {{- if eq (default "cookie" .Values.sessionStorage.type) "redis" }}
         - name: OAUTH2_PROXY_SESSION_STORE_TYPE
           value: "redis"
-        {{- if or .Values.sessionStorage.redis.password .Values.sessionStorage.redis.existingSecret }}
+         {{- if or .Values.sessionStorage.redis.existingSecret .Values.sessionStorage.redis.password (and .Values.redis.enabled (.Values.redis.auth).enabled )}}
```

Comment:
```suggestion
        {{- if or .Values.sessionStorage.redis.password .Values.sessionStorage.redis.existingSecret (and .Values.redis.enabled .Values.redis.auth.enabled )}}
```

Reply:
I had it without the extra parenthesis as the beginning, but this is causing issues with Redis is not enabled. Helm errors on the evaluation of `.Values.redis.auth.enabled` for some reason, if the redis chart is not enabled. I would assume it stops early to evaluate when using the, and it's not the case, it seems.